aws/resource_aws_elasticache_cluster_test.go

```diff
@@ -370,7 +370,7 @@ func TestAccAWSElasticacheCluster_vpc(t *testing.T) {
 					testAccCheckAWSElasticacheClusterExists("aws_elasticache_cluster.bar", &ec),
 					testAccCheckAWSElasticacheClusterAttributes(&ec),
 					resource.TestCheckResourceAttr(
-						"aws_elasticache_cluster.bar", "availability_zone", "us-west-2a"),
+						"aws_elasticache_cluster.bar", "availability_zone", "us-east-1a"),
```

Comment:
These changes appear to be unrelated. 😉

Reply:
Yes, but I was unable to get the tests to pass before or after my changes without this one. Is this something that should go in a separate PR though?

Reply:
Yes, please! Its likely the parallel testing issue we talked about earlier today where one of the resource configurations has `provider "aws"` with us-east-1 region declared (or a acceptance test function is monkeying with the `AWS_DEFAULT_REGION` environment variable for EC2-Classic reasons), which works fine in TeamCity as that environment runs each test in separate `go test` processes, but not locally where the test provider is shared in the same `go test` process. We'll eventually clean up the oddball `us-east-1` testing setups as part of a larger project with a more elegant solution.

If you do see hardcoded availability zones and such, the best course of action with them is instead using the [`aws_availability_zones` data source](https://www.terraform.io/docs/providers/aws/d/availability_zones.html), e.g.

```hcl
data "aws_availability_zones" "available" {
  state = "available"
}

# ... other configuration ...

resource "aws_subnet" "foo" {
    # ... other configuration ...
    availability_zone = "${data.aws_availability_zones.available.names[0]}"
}
```

And the checks within the acceptance test can use something like the below instead of hardcoding the AZ:

```go
resource.TestCheckResourceAttrPair("aws_elasticache_cluster.bar", "availability_zone", "data.aws_availability_zones.available", "names.0"),
```